### PR TITLE
KEP-3619: implement RuntimeStatus.features.supplemental_groups_policy

### PR DESCRIPTION
**Pull request by everpeace**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add an implementation of `RuntimeStatus.features.supplemental_groups_policy` to CRI-O so that Kubernetes users can recognize if the CRI implementation supports SupplementalGroupsPolicy.

This also helps CI tests can gracefully skip if the CRI implementation does not support SupplementalGroupsPolicy. 

Related to:

- #8268

For:
- https://github.com/kubernetes/enhancements/issues/3619

Depends on:
- https://github.com/kubernetes/kubernetes/pull/125470

Depended by:
- https://github.com/kubernetes-sigs/cri-tools/pull/1438

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

I'm aiming to alpha-release this feature in Kubernetes 1.31 (expected release date is 13th August 2024). Thank you, in advance 🙇🙇 

#### Does this PR introduce a user-facing change?

```release-note
Add RuntimeStatus.features.supplemental_groups_policy field (KEP-3619)
```

### CODE REVIEW

**Review by everpeace — COMMENTED** [review]
@mrunalp and cri-o reviwers

could you help review??

And, I would be glad if you help retrying failing jobs. I believe those failure is not related to this PR.

**Comment by kwilczynski** [comment]
@cri-o/cri-o-maintainers, please have a look. Thank you!

File: go.mod [hunk 1]
```diff
@@ -77,7 +77,7 @@ require (
 	k8s.io/api v0.31.0-alpha.3
 	k8s.io/apimachinery v0.31.0-alpha.3
 	k8s.io/client-go v0.31.0-alpha.3
-	k8s.io/cri-api v0.31.0-alpha.3
+	k8s.io/cri-api v0.31.0-beta.0.0.20240716205706-865479a3e1b3
```
Comment (saschagrunert) [inline go.mod:80]:
```suggestion
	k8s.io/cri-api v0.31.0-beta.0
```

Comment (everpeace) [reply go.mod:80]: @saschagrunert Sorry, unfortunately my k/k PR missed v1.31.0-beta.0😿 So, I need the latest version.

This is a commit in cri-api repo: https://github.com/kubernetes/cri-api/commit/865479a3e1b3625494674397b1abc5f72702ca2e (this is not included in the `0.31.0-beta.0` tag)

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**